extensions/macos-utilities/src/terminal-history.tsx

```diff
@@ -0,0 +1,133 @@
+import { ActionPanel, Action, Icon, Color, List, confirmAlert, Alert, showHUD } from "@raycast/api";
+import { useEffect, useState } from "react";
+import { execSync } from "child_process";
+import os from "os";
+import { createRaySoLink, getHistory, restoreHistory, saveFile } from "./utils";
+
+let history: string[] = [];
+
+export default function Command() {
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(false);
+  }, [loading]);
+
+  history = getHistory();
```

Comment:
I don't think this does quite what might be intended.

Firstly, the `loading` variable isn't hooked up to the list's `isLoading` prop, so I don't _think_ it's actually doing anything.

Secondly though, because `getHistory` is synchronous, it'll run on every render of this component (which is typically very often). The loading state will start in `true` for the first render. The first render will then kick off the effect, which will set it to `false`, which will _coincidentally_ only happen after the history has been read, and then the component will be re-rendered with the `false` state. This will then reload the history again, but without the loading state.

I'd suggest a) investigating `usePromise` and the `isLoading` parameter from that, and b) making `getHistory` async so that it doesn't block the JS execution thread and lock up Raycast. These two things should improve the responsiveness, ensure the history is only loaded once rather than the current two or more times, and also ensure the loading state is actually correct.

Reply:
i went with the approach of using `useState` variables that way we don't have an overly abundant usecase of doing the syncronous approach of `getHistory`.